### DISCUSSION TITLE
Consistency in use of `source` in docker images

### DISCUSSION THREAD

**Discussion by ivanperez-keera** [space-ros/docker#106] [Ideas]
This is a minor thing, but, as I was going through the image, I noticed that these two `source` invocations are written in different ways:

https://github.com/space-ros/docker/blob/76cd412a7379dd5a5ec338b7b75659afbbf0c9a1/space_robots/Dockerfile#L80-L87

It'll help us catch errors and keep things well maintained if we are consistent. I think we should pick either this way of doing it:

https://github.com/space-ros/docker/blob/76cd412a7379dd5a5ec338b7b75659afbbf0c9a1/space_robots/Dockerfile#L81-L82

or this way:

https://github.com/space-ros/docker/blob/76cd412a7379dd5a5ec338b7b75659afbbf0c9a1/space_robots/Dockerfile#L86

I personally like the former because I don't like long lines. I find it much harder to catch mistakes in long lines. Also, editors can break lines but github doesn't, and it leads to `git diff`s that are harder to process.

**Comment by ivanperez-keera** [comment]
This is minor and perhaps not a priority. I'm filing it here first to see what people think. If we want to do something about it, we can promote this discussion to an issue.

**Comment by Bckempa** [comment]
I don't think those are equivalent. In the first version the overlays are sourced in subprocesses that are immediately discarded, while in the second version the Space ROS and Move It 2 workspaces are sourced when `rosdep` is called.

**Reply by ivanperez-keera** [reply]
I didn't notice the end `'`. If what you are saying is right, do the first two have no effect then?

**Reply by Bckempa** [reply]
Correct. I haven't tested it yet, but I assume it would impact the `rosdep` call because `--ignore-src` will skip installing any package found in the `ROS_PACKAGE_PATH` which would be modified by sourcing the two workspaces.

**Reply by Bckempa** [reply]
I'll confirm there is an impact and convert this to a bug if necessary.
